lib/src/package_analysis/package_analysis.dart

```diff
@@ -21,50 +29,348 @@ class SummaryCommand extends Command {
         .exists()) {
       throw StateError(
           'Run `dart pub get` to fetch dependencies before analysing this package.');
     }
 
-    var collection =
-        AnalysisContextCollection(includedPaths: [packageLocation]);
+    final session = AnalysisContextCollection(includedPaths: [packageLocation])
+        .contextFor(packageLocation)
+        .currentSession;
+    final packageAnalysisContext = PackageAnalysisContextWithStderr(
+      session: session,
+      packagePath: packageLocation,
+    );
+
+    final packageShape = await summarizePackage(
+      context: packageAnalysisContext,
+      packagePath: packageLocation,
+    );
+
+    final packageJson = packageShape.toJson();
 
-    var packageJson = (await summarizePackage(
-      _PackageAnalysisContext(collection),
-      packageLocation,
-    ))
-        .toJson();
-    var indentedEncoder = const JsonEncoder.withIndent('  ');
     print(indentedEncoder.convert(packageJson));
   }
 }
 
-class _PackageAnalysisContext extends PackageAnalysisContext {
+class LowerBoundConstraintAnalysisCommand extends Command {
   @override
-  late final AnalysisContextCollection analysisContextCollection;
+  final name = 'lbcanalysis';
+  @override
+  final description = 'Performs lower bound analysis on a single package.';
+  @override
+  final usage = '''Required positional arguments:
+  1) path to a directory containing package version list cache
+  2) name of the package to analyze''';
+
+  LowerBoundConstraintAnalysisCommand();
+
+  @override
+  Future<void> run() async {
+    final arguments = argResults!.rest;
+    if (arguments.length != 2) {
+      throw ArgumentError(
+          'This command accepts exactly 2 positional arguments.');
+    }
 
-  _PackageAnalysisContext(AnalysisContextCollection contextCollection) {
-    analysisContextCollection = contextCollection;
+    final cachePath = arguments[0];
+    final targetName = arguments[1];
+    if (!(await Directory(cachePath).exists())) {
+      throw ArgumentError(
+          'The directory containing package version list cache could not be found.');
+    }
+
+    // create a unique temporary directory for the target and the dependencies
+    final tempDir = await Directory(Directory.systemTemp.path)
+        .createTemp('lbcanalysis_temp');
+
+    try {
+      final foundIssues = await lowerBoundConstraintAnalysis(
+        targetName: targetName,
+        tempPath: path.canonicalize(tempDir.path),
+        cachePath: cachePath,
+      );
+
+      for (final issue in foundIssues) {
+        final c = http.Client();
+        late final String targetHomepage;
+        late final String targetRepository;
+        late final String dependencyHomepage;
+        late final String dependencyRepository;
+        // TODO: not sure if nested try-finally here is OK
+        // I'm doing this because usually there are no found issues and no need to create an http client
+        try {
```

Comment:
not sure if nested try-finally here is OK
I'm doing this because usually there are no found issues and no need to create an http client

Reply:
close client in outer finally